src/utils.ts

```diff
@@ -179,27 +176,53 @@ export function tokenToPythSymbol(token: string) {
   return undefined;
 }
 
-export async function getPythPrice(splMint: PublicKey) {
+export async function getPythPrice(baseMint: PublicKey) {
   const connection: Connection = new Connection(API_URL);
   const pythPublicKey = getPythProgramKeyForCluster(IS_DEV ? 'devnet': 'mainnet-beta');
   const pythClient = new PythHttpClient(connection, pythPublicKey);
   const data = await pythClient.getData();
 
   for (let symbol of data.symbols) {
     const price = data.productPrice.get(symbol)!;
-    if (tokenToPythSymbol(splMintToToken(splMint)) == symbol) {
+    if (tokenToPythSymbol(splMintToToken(baseMint)) == symbol) {
       return price.price;
     }
   }
   return 0.0;
 }
 
-export function tokenType(type: string) {
-  if (type == "Upside") {
-    return "call";
+export function getDIPType(dip_deposit: DIPDeposit) {
+  for (let premium of premiumAssets){
+    if (dip_deposit.quoteAsset == premium) {
+      return "call";
+    }
+    if (dip_deposit.baseAsset == premium) {
+      return "put";
+    }
+  }
+  return 'undefined';
+}
+
+export function getDIPDirection(dip_deposit: DIPDeposit) {
+  for (let premium of premiumAssets){
+    if (dip_deposit.quoteAsset == premium) {
```

Comment:
is this functionally different from getDIPType?

Reply:
Just return Upside or Downside conform to this from the API
SPL_TOKEN,PREMIUM_ASSET,EXPIRATION,STRIKE,TYPE,POOL,STYLE,SETTLEMENT
Example:
BTC,USDC,2022-07-08,22000,UPSIDE,DIP,A,P

Reply:
Ah this will break threshold price, updating

Reply:
Should have the logic of determining direction in one just one place. Will help with code maintainability

Reply:
Should this sit someplace in the DIP repo? DIP state could be read inclusive of getDIPType & adjustStrike?